### PR TITLE
Add Helm chart

### PR DESCRIPTION
**Pull request by amrecio**
This PR adds a Helm chart in `deploy/chart`.

### CODE REVIEW

File: deploy/chart/README.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+- Kubernetes 1.12+ with Beta APIs enabled
+
+## Installing the Chart
+
+To install the chart with the release name `my-release`:
+
+```console
+$ git clone https://github.com/rancher/local-path-provisioner.git
+$ helm install --name my-release ./deploy/chart/local-path-provisioner
```
Comment (yasker) [inline deploy/chart/README.md]: Should be something like
```
$ git clone https://github.com/rancher/local-path-provisioner.git
$ cd local-path-provisioner
$ helm install --name local-path-storage ./deploy/chart/
```

File: deploy/chart/templates/NOTES.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+You can provision an EFS-backed persistent volume with a persistent volume claim like below:
```
Comment (yasker) [inline deploy/chart/templates/NOTES.txt]: What's this?

Comment (amrecio) [reply deploy/chart/templates/NOTES.txt]: Actually, that's a leftover from a different internal chart that slipped by. I'll change it.

File: deploy/chart/README.md (hunk 1 above)
Comment (amrecio) [reply deploy/chart/README.md]: Good point.

Comment (yasker) [reply deploy/chart/README.md]: Unsolved?

**Comment by yasker** [comment]
Also, can you merge the commits into one? No need to make separate commits since I prefer a clean git history and I can check the change manually.

**Comment by amrecio** [comment]
Sure. I've squashed the commits as requested.

File: deploy/chart/README.md (hunk 1 above)
Comment (amrecio) [reply deploy/chart/README.md]: Sorry, I didn't notice the path was wrong too. Fixed now.

File: deploy/chart/README.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+This chart bootstraps a [Local Path Provisioner](https://github.com/rancher/local-path-provisioner) deployment on a
+[Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.
+
+## Prerequisites
+
+- Kubernetes 1.12+ with Beta APIs enabled
+
+## Installing the Chart
+
+To install the chart with the release name `my-release`:
```
Comment (yasker) [inline deploy/chart/README.md]: `local-path-storage`

File: deploy/chart/README.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+```
+
+The command deploys Local Path Provisioner on the Kubernetes cluster in the default configuration. The
+[configuration](#configuration) section lists the parameters that can be configured during installation.
+
+> **Tip**: List all releases using `helm list`
+
+## Uninstalling the Chart
+
+To uninstall/delete the `my-release` deployment:
```
Comment (yasker) [inline deploy/chart/README.md]: `local-path-storage`

File: deploy/chart/README.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+```console
+$ helm delete my-release
```
Comment (yasker) [inline deploy/chart/README.md]: `$ helm delete --purge local-path-storage`

File: deploy/chart/README.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+- Kubernetes 1.12+ with Beta APIs enabled
+
+## Installing the Chart
+
+To install the chart with the release name `my-release`:
+
+```console
+$ git clone https://github.com/rancher/local-path-provisioner.git
+$ cd local-path-provisioner
+$ helm install --name local-path-storage ./deploy/chart/
```
Comment (yasker) [inline deploy/chart/README.md]: Better to keep consistent with yaml deployment.
```
helm install ./deploy/chart/ --name local-path-storage --namespace local-path-storage
```

File: deploy/chart/README.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+`hostPath` based persistent volume on the node automatically. It utilizes the features introduced by Kubernetes [Local
+Persistent Volume feature](https://kubernetes.io/blog/2018/04/13/local-persistent-volumes-beta/), but make it a simpler
+solution than the built-in `local` volume feature in Kubernetes.
+
+## TL;DR;
+
+```console
+$ git clone https://github.com/rancher/local-path-provisioner.git
+$ cd local-path-provisioner
+$ helm install ./deploy/chart/
```
Comment (yasker) [inline deploy/chart/README.md]:
```
helm install ./deploy/chart/ --name local-path-storage --namespace local-path-storage
```

File: deploy/chart/README.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+| `serviceAccount.name`               | Name of the Local Path Provisioner service account to use or create             | `nil`                                                                               |
+| `nodeSelector`                      | Node labels for Local Path Provisioner pod assignment                           | `{}`                                                                                |
+| `tolerations`                       | Node taints to tolerate                                                         | `[]`                                                                                |
+| `affinity`                          | Pod affinity                                                                    | `{}`                                                                                |
+
+Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example,
+
+```console
+$ helm install local-path-storage --name local-path-provisioner \
```
Comment (yasker) [inline deploy/chart/README.md]:
```
helm install ./deploy/chart/ --name local-path-storage --namespace local-path-storage --set storageClass.provisionerName=rancher.io/local-path
```

File: deploy/chart/README.md [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+  --set storageClass.provisionerName=rancher.io/local-path
+```
+
+Alternatively, a YAML file that specifies the values for the above parameters can be provided while installing the
+chart. For example,
+
+```console
+$ helm install ./deploy/chart/ --name local-path-storage -f values.yaml
```
Comment (yasker) [inline deploy/chart/README.md]:
```
helm install ./deploy/chart/ --name local-path-storage --namespace local-path-storage -f values.yaml
```

File: deploy/chart/README.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+```
+
+If the output contains "beta", you may install the chart with RBAC enabled (see below).
+
+### Enable RBAC role/rolebinding creation
+
+To enable the creation of RBAC resources (On clusters with RBAC). Do the following:
+
+```console
+$ helm install --name local-path-storage ./deploy/chart/ --set rbac.create=true
```
Comment (yasker) [inline deploy/chart/README.md]:
```
helm install ./deploy/chart/ --name local-path-storage --namespace local-path-storage --set rbac.create=true
```

**Comment by yasker** [comment]
@meldafrawi Can you help to verify this chart?

**Comment by meldafrawi** [comment]
verified.

**Pull request merged**